vtex/faststore
# Feat: Adds `Carousel` Component 782

## What's the purpose of this pull request?

`faststore/components`
- [x] Moves `Carousel` component;
- [x] Adds `Carousel` feature doc;

`faststore/ui`
- [x] Adds `Carousel` styles and fixes;
- [x] Removes `Bullets` component (includes it on `Carousel`)
- [x] Removes `Arrows`

`faststore/core`
- [x] Review `Carousel` structure;
- [x] Applies and Reviews `Carousel` usage on `ProductShelf`
- [x] Delete stories and styles;

`docs`
- [x] Updates `ProductShelf` doc with examples;
- [x] Updates `ProductGrid` doc example

## How to test it?

https://user-images.githubusercontent.com/3356699/234901770-ecb0467c-9c86-4fec-b0fa-2e14387af5a6.mov


1. Check the `Carousel` component using the doc preview link:
- [Carousel](https://faststore-site-git-feat-carousel-fs-782-faststore.vercel.app/components/molecules/carousel)
- You can also check the examples on [ProductShelf](https://faststore-site-git-feat-carousel-fs-782-faststore.vercel.app/components/organisms/product-shelf)

2. run `yarn dev` on `faststore/core` application and check the `ProductShelf`(this component is current using the `Carousel` occurrence on Homepage:
- Check it on mobile version, you might see the carousel working as expected:

https://user-images.githubusercontent.com/3356699/234902338-d4344365-d6f7-44f2-ac61-ce7adb29bf20.mov


### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/52
